pkg/ddevapp/commands.go

```diff
@@ -1,47 +1,69 @@
 package ddevapp
 
 import (
-	"github.com/ddev/ddev/pkg/fileutil"
-	"github.com/ddev/ddev/pkg/globalconfig"
-	"github.com/ddev/ddev/pkg/util"
-	copy2 "github.com/otiai10/copy"
+	"fmt"
 	"os"
+	"path"
 	"path/filepath"
+
+	dockerImages "github.com/ddev/ddev/pkg/docker"
+	"github.com/ddev/ddev/pkg/dockerutil"
+	"github.com/ddev/ddev/pkg/globalconfig"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/util"
 )
 
-// PopulateCustomCommandFiles sets up the custom command files in the project
+// PopulateGlobalCustomCommandFiles sets up the custom command files in the project
 // directories where they need to go.
-func PopulateCustomCommandFiles(app *DdevApp) error {
-
+func PopulateGlobalCustomCommandFiles() error {
 	sourceGlobalCommandPath := filepath.Join(globalconfig.GetGlobalDdevDir(), "commands")
 	err := os.MkdirAll(sourceGlobalCommandPath, 0755)
 	if err != nil {
 		return nil
 	}
 
-	projectCommandPath := app.GetConfigPath("commands")
-	// Make sure our target global command directory is empty
-	copiedGlobalCommandPath := app.GetConfigPath(".global_commands")
-	err = os.MkdirAll(copiedGlobalCommandPath, 0755)
+	// Remove contents of the directory, if the directory exists and has some contents
+	commandDirInVolume := "/mnt/ddev-global-cache/global-commands/"
+	_, stderr, err := performTaskInContainer([]string{"sh", "-c", "rm -r " + path.Join(commandDirInVolume, "*") + " || true"})
 	if err != nil {
-		util.Error("Unable to create directory %s: %v", copiedGlobalCommandPath, err)
-		return nil
+		return fmt.Errorf("unable to rm %s: %v (stderr=%s)", commandDirInVolume, err, stderr)
 	}
 
-	// Make sure it's empty
-	err = fileutil.PurgeDirectory(copiedGlobalCommandPath)
+	// Copy commands into container (this will create the directory if it's not there already)
+	uid, _, _ := util.GetContainerUIDGid()
+	err = dockerutil.CopyIntoVolume(sourceGlobalCommandPath, "ddev-global-cache", "global-commands", uid, "host", false)
 	if err != nil {
-		util.Error("Unable to remove %s: %v", copiedGlobalCommandPath, err)
-		return nil
+		return err
 	}
 
-	err = copy2.Copy(sourceGlobalCommandPath, copiedGlobalCommandPath)
+	// Make sure all commands can be executed
+	_, stderr, err = performTaskInContainer([]string{"sh", "-c", "find " + commandDirInVolume + " -type f -exec chmod 0755 {} \\;"})
```

Comment:
I don't think we need a find here; in there it's not a problem that we might be changing modes on a few things that don't matter. This should be a little faster.

```suggestion
	_, stderr, err = performTaskInContainer([]string{"sh", "-c", "chmod -R u+rwx " + commandDirInVolume})
```

Reply:
Done